torchtitan/datasets/hf_datasets.py

```diff
@@ -88,15 +88,15 @@ def __init__(
         ds = dataset_loader(path)
 
         self.dataset_name = dataset_name
-        self._data = split_dataset_by_node(ds, rank, world_size)
+        self._data = split_dataset_by_node(ds, dp_rank, dp_world_size)
```

Comment:
nice, this is indeed needed. When tracing the code.. I need to double check its definition in the caller, although the comment specified that.